homeassistant/components/bmw_connected_drive/quality_scale.yaml

```diff
@@ -0,0 +1,80 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      Does not have custom services
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions:
+    status: exempt
+    comment: |
+      Does not have custom services
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup:
+    status: exempt
+    comment: |
+      This integration doesn't have any events.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions:
+    status: exempt
+    comment: |
+      Does not have custom services
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates: done
+  reauthentication-flow: done
+  test-coverage: done
```

Comment:
We should not touch internals in the tests. So for example with `test_update_success`, we should not touch `entry.runtime_data.coordinator`, but we should assert what that actually means. So we should pick an entity and check if that has been updated correctly. At the end of the day, I don't really care if `last_update_success` is True, I want to see if my entities expose the right thing.

Same for the other tests, if last_update_success is False, we should assert that entities are unavailable.

But since we're touching entities, we should move those tests to the entity platforms instead of having them in test_coordinator.py

Reply:
Fixed by #133704.